File: Java/src/main/java/via/sep3/group2/models/OrderLineDTO.java
package via.sep3.group2.models;

import javax.persistence.*;

@Entity
@Table(name="orderline")
public class OrderLineDTO {
    @EmbeddedId
    OrderLineKey id;

    @ManyToOne
    @MapsId("orderserial")
    @JoinColumn(name="serialorder"    )
    OrdersDTO ordersDTO;

    @ManyToOne
    @MapsId("bookisbn")
    @JoinColumn(name="isbn",
            foreignKey = @ForeignKey(
                    name="isbn",
                    foreignKeyDefinition = "FOREIGN KEY (isbn) REFERENCES book(isbn) ON UPDATE CASCADE ON DELETE CASCADE ")
    )
    BookDTO bookDTO;

    private int qte;

}

File: CSharp/Shared/OrderLineDTO.cs
namespace ModelClasses;

public class OrderLineDTO
{
    public long? SerialOrder { get; set; }
    public string Isbn { get; set; } 
    public int Quantity { get; set; }

    public Book? book { get; set; } 

    public OrderLineDTO(string isbn, int quantity)
    {
        Isbn = isbn;
        Quantity = quantity;
       
    }

    public OrderLineDTO(long? serialOrder, string isbn, int quantity)
    {
        SerialOrder = serialOrder;
        Isbn = isbn;
        Quantity = quantity;
        
    }

    public OrderLineDTO(Book book)
    {
        this.book = book;
    }

    public OrderLineDTO(Book book, long? serialOrder, string isbn, int quantity)
    {
        this.book = book;
        SerialOrder = serialOrder;
        Isbn = isbn;
        Quantity = quantity;
    }

    public OrderLineDTO()
    {
    }
}